dtfs-central-api/src/services/tfm/deal-cancellation.service.ts

```diff
@@ -30,9 +52,54 @@ export class DealCancellationService {
     };
 
     if (isDealCancellationPastOrPresent) {
-      return await TfmDealCancellationRepo.submitDealCancellation({ dealId, cancellation, activity, auditDetails });
+      const response = await TfmDealCancellationRepo.submitDealCancellation({ dealId, cancellation, activity, auditDetails });
```

Comment:
I do think the method I suggested in a previous comment would reduce this complexity... So this would be

```
// rename to processCancellationRequest
    const dealWithCancellationAmendment = await AmendmentController.addCancellationAmendment()
    if(isFuture) {
        // Cron job will process this when needed 
        return dealWithCancellationAmendment
    }
    await TfmDealCancellationRepo.cancelDeal() // or whatever -- this goes to the same one the cron job does
    return await TfmDealCancellationRepo.scheduleDealCancellation({ dealId, cancellation, activity, auditDetails });
```

which would reduce a lot of pain?

Reply:
I'm not sure I quite follow this - will revist on Monday with you :)

Reply:
When the deal cancellation is in the future we want
- activity and comments page entry
- cancellation object updating (status = scheduled)
- return facility & deal details for sending email

When the deal cancellation is in the past we want
- the above (but cancellation object status cancelled)
- facility and deal status update in tfm
- facility and deal status in portal

When the cron job activates we want
- facility and deal status update in tfm
- facility and deal status in portal


I guess that could be divided up into two functions, with scheduling calling one, cron job calling the other and effective date in past calling both, this would make it a bit more dry but at the cost of making more DB calls - I generally think we want to avoid this (since no transactions in mongo) but keen to hear thoughts

Reply:
We had a look at this -- as it's not expressively part of this review we're leaving this for now, but I've had a chat with Mark about ways we can improve in future